include/seqan3/io/stream/parse_condition.hpp

```diff
@@ -0,0 +1,468 @@
+ * \ingroup io
+ */
+template <parse_condition_concept... condition_ts>
+//\cond
+    requires sizeof...(condition_ts) >= 2
+//\endcond
+struct or_fn : public parse_condition<or_fn<condition_ts...>>
+{
+    //!\brief The message representing the disjunction of the associated conditions.
+    static constexpr auto msg = detail::condition_message_v<'|', condition_ts...>;
+
+    //!\brief Calls each parse condition with `val` and returns the logical disjunction of their results.
+    template <typename value_t>
+    inline bool operator()(value_t && val) const
+    {
+        return (... || std::invoke(condition_ts(), std::forward<value_t>(val)));
+    }
+};
+
+/*!\brief Logical conjunction operator for parse conditions.
+ * \implements seqan3::detail::parse_condition_concept
+ * \tparam condition_ts Template parameter pack over all parse condition types. Must contain at least 2 template parameters.
+ *                      Must satisfy the seqan3::detail::parse_condition_concept.
+ * \ingroup io
+ */
+template <parse_condition_concept... condition_ts>
+//\cond
+    requires sizeof...(condition_ts) >= 2
+//\endcond
+struct and_fn : public parse_condition<and_fn<condition_ts...>>
+{
+    //!\brief The message representing the conjunction of the associated conditions.
+    static constexpr auto msg = detail::condition_message_v<'&', condition_ts...>;
+
+    //!\brief Calls each parse conditions with `val` and returns the logical conjunction.
+    template <typename value_t>
+    inline bool operator()(value_t && val) const
+    {
+        return (... && std::invoke(condition_ts(), std::forward<value_t>(val)));
+    }
+};
+
+/*!\brief Logical not operator for a parse condition.
+ * \implements seqan3::detail::parse_condition_concept
+ * \tparam condition_t Template parameter to apply the not-operator for.
+ *                     Must satisfy the seqan3::detail::parse_condition_concept.
+ * \ingroup io
+ */
+template <parse_condition_concept condition_t>
+struct not_fn : public parse_condition<not_fn<condition_t>>
+{
+    //!\brief The message representing the negation of the associated condition.
+    static constexpr auto msg = constexpr_string{'!'} + condition_t::msg;
+
+    //!\brief Calls the stored parse condition with `val` and returns the negated result.
+    template <typename value_t>
+    inline bool operator()(value_t && val) const
+    {
+        return !std::invoke(condition_t(), std::forward<value_t>(val));
+    }
+};
+
+// ----------------------------------------------------------------------------
+// is_in_range
+// ----------------------------------------------------------------------------
+
+/*!\brief Parse condition that checks if a given value is in the range of `rng_beg` and `rng_end`.
+ * \implements seqan3::detail::parse_condition_concept
+ * \tparam rng_beg non-type template parameter denoting the begin of the allowed range. Must be less or greater to `rng_end`.
```

Comment:
you mean less or _**equal**_ probably... 
to ==> than

Reply:
probably 🤣

Reply:
In fact it should be `is less than or equal to` 🤓